File: app/src/main/java/com/mcal/worldcraft/util/RandomUtil.java
package com.mcal.worldcraft.util;

import com.mcal.worldcraft.math.MathUtils;

public class RandomUtil {
    public static int getRandomInRangeInclusive(int min, int max) {
        return ((int) (Math.random() * ((max - min) + 1))) + min;
    }

    public static int getRandomSignedInRangeInclusive(int min, int max) {
        return getRandomInRangeInclusive(min, max) * MathUtils.getRandomSign();
    }

    public static float getRandomInRangeExclusive(float min, float max) {
        return ((float) (Math.random() * (max - min))) + min;
    }

    public static float getRandomSignedInRangeExclusive(float min, float max) {
        return getRandomInRangeExclusive(min, max) * MathUtils.getRandomSign();
    }

    public static boolean getChance(float value) {
        return Math.random() <= ((double) value);
    }
}

File: app/src/main/java/com/mcal/worldcraft/utils/RandomUtil.kt
package com.mcal.worldcraft.utils

import com.mcal.worldcraft.math.MathUtils

object RandomUtil {
    @JvmStatic
    fun getRandomInRangeInclusive(min: Int, max: Int): Int {
        return (Math.random() * (max - min + 1)).toInt() + min
    }

    @JvmStatic
    fun getRandomSignedInRangeInclusive(min: Int, max: Int): Int {
        return getRandomInRangeInclusive(min, max) * MathUtils.getRandomSign()
    }

    @JvmStatic
    fun getRandomInRangeExclusive(min: Float, max: Float): Float {
        return (Math.random() * (max - min)).toFloat() + min
    }

    @JvmStatic
    fun getRandomSignedInRangeExclusive(min: Float, max: Float): Float {
        return getRandomInRangeExclusive(min, max) * MathUtils.getRandomSign()
    }

    @JvmStatic
    fun getChance(value: Float): Boolean {
        return Math.random() <= value.toDouble()
    }
}